core/src/main/java/io/jenkins/pluginhealth/scoring/probes/SecurityScanGithubWorkflowProbe.java

```diff
@@ -0,0 +1,119 @@
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+@Component
+@Order(SecurityScanGithubWorkflowProbe.ORDER)
+public class SecurityScanGithubWorkflowProbe extends Probe {
+    public static final int ORDER = LastCommitDateProbe.ORDER + 100;
+    public static final String KEY = "security-scan";
+    public static final String SEARCH_LINE = "jenkins-infra/jenkins-security-scan/.github/workflows/jenkins-security-scan.yaml";
+    private static final String WORKFLOWS_DIRECTORY = ".github/workflows";
+    private static final Logger LOGGER = LoggerFactory.getLogger(SecurityScanGithubWorkflowProbe.class);
+
+    @Override
+    protected ProbeResult doApply(Plugin plugin, ProbeContext context) {
+        final Path repository = context.getScmRepository();
+        final Path workflowPath = Paths.get(repository.toString(), WORKFLOWS_DIRECTORY);
+
+        if (! Files.exists(workflowPath)) {
+            return ProbeResult.failure(key(), "GitHub workflow directory could not be found in the plugin");
+        }
+
+            try (Stream<Path> files = Files.find(workflowPath, 1,
+                (path, basicFileAttributes) -> Files.isRegularFile(path)
+            )) {
+                final ObjectMapper yaml = new ObjectMapper(new YAMLFactory());
+
+                return files
+                    .map(file -> {
+                        try {
+                            return yaml.readValue(Files.newInputStream(file), SecurityScanGithubWorkflowProbe.WorkflowDefinition.class);
+                        } catch (IOException e) {
+                            LOGGER.error("Couldn't not read {} for {} on {}", file, key(), plugin.getName(), e);
+                            return new SecurityScanGithubWorkflowProbe.WorkflowDefinition(Map.of());
+                        }
+                    })
+                    .filter(wf -> wf.jobs() != null && !wf.jobs().isEmpty())
+                    .flatMap(wf -> wf.jobs().values().stream())
+                    .map(SecurityScanGithubWorkflowProbe.WorkflowJobDefinition::uses)
+                    .filter(Objects::nonNull)
+                    .anyMatch(def -> def.startsWith(SEARCH_LINE)) ?
+                    ProbeResult.success(key(), "GitHub workflow security scan is configured in the plugin") :
+                    ProbeResult.failure(key(), "GitHub workflow security scan is not configured in the plugin");
+
+        } catch (IOException e) {
+            return ProbeResult.error(key(), e.getMessage());
+        }
+    }
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    private record WorkflowJobDefinition(String uses) {
```

Comment:
This deserves Javadoc, to explain what that `record` is for, and ideally with a link to the syntax reference of GitHub workflows, so future developers understand where `uses` comes from.

Reply:
Not to worry about those right now as they should be de-duplicated first.